src/scripts/main.js

```diff
@@ -1,3 +1,247 @@
 'use strict';
 
-// write code here
+/* eslint-disable */
+const body = document.querySelector('body');
+const tHeadCells = document.querySelectorAll('table thead th');
+const tbody = document.querySelector('table tbody');
+const rows = [...tbody.querySelectorAll('tr')];
+
+const officeOptions
+  = ['Tokyo', 'Singapore', 'London', 'New York', 'Edinburgh', 'San Francisco'];
+
+employeeForm();
+
+for (let i = 0; i < tHeadCells.length; i++) {
+  tHeadCells[i].addEventListener('click', () => {
+    const receivedRows = sortByIdAndType(i);
+    receivedRows.forEach(row => tbody.append(row));
+  });
+}
+
+function sortByIdAndType(index) {
+  const currentRows = [...tbody.querySelectorAll('tr')];
+  const rows = [...currentRows];
+
+  rows.sort((rowA, rowB) => {
+    const cellA = rowA.cells[index].innerText;
+    const cellB = rowB.cells[index].innerText;
+
+    return cellA.localeCompare(cellB, 'en', { numeric: true });
+  });
+
+  if (JSON.stringify(rows.map(row => row.innerText))
+    === JSON.stringify(currentRows.map(row => row.innerText))) {
+    rows.sort((rowA, rowB) => {
+      const cellA = rowA.cells[index].innerText;
+      const cellB = rowB.cells[index].innerText;
+
+      return cellB.localeCompare(cellA, 'en', { numeric: true });
+    });
+  }
+  return rows;
+}
+
+tbody.addEventListener('click', (event) => {
+  if (!event.target.parentElement.tagName === 'TR') {
+    return;
+  }
+
+  const activeClasses = tbody.querySelectorAll('.active');
+  for (const activeClass of activeClasses) {
```

Comment:
Replace for of loop with forEach, e.g. activeClasses.forEach(activeClass => activeClass.classList.remove('active'));